File: tkpdinbox/src/main/java/com/tokopedia/inbox/inboxchat/adapter/DummyAdapter.java
package com.tokopedia.inbox.inboxchat.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.tokopedia.inbox.R;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by stevenfredian on 9/18/17.
 */

public class DummyAdapter extends RecyclerView.Adapter<DummyAdapter.ViewHolder>{

    private List<Integer> list;

    public DummyAdapter(ArrayList<Integer> objects) {
        this.list = objects;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.dummy, parent, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        Integer c = list.get(position);
    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    public List<Integer> getList() {
        return list;
    }

    public void setList(ArrayList<Integer> local) {
        this.list = local;
    }

    public class ViewHolder extends RecyclerView.ViewHolder{

        public ViewHolder(View itemView) {
            super(itemView);
        }
    }

}

File: features/merchant/seller_mvc_creation/src/main/java/com/tokopedia/mvc/presentation/list/adapter/DummyAdapter.kt
package com.tokopedia.mvc.presentation.list.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.tokopedia.kotlin.extensions.view.ZERO
import com.tokopedia.mvc.databinding.SmvcDummyBinding

class DummyAdapter: RecyclerView.Adapter<DummyAdapter.CriteriaViewHolder>() {

    private var data: List<String> = emptyList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CriteriaViewHolder {
        val binding = SmvcDummyBinding.inflate(LayoutInflater.from(parent.context),
            parent, false)
        return CriteriaViewHolder(binding)
    }

    override fun getItemCount() = data.size

    override fun onBindViewHolder(holder: CriteriaViewHolder, position: Int) {
        data.getOrNull(position)?.let { menu ->
            holder.bind(menu)
        }
    }

    fun setDataList(newData: List<String>) {
        data = newData
        notifyItemRangeChanged(Int.ZERO, newData.size)
    }

    inner class CriteriaViewHolder(private val binding: SmvcDummyBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(title: String) {
            val numberText = "${adapterPosition.inc()}."
            binding.tvText.text = numberText + title
        }
    }
}
